### PR TITLE
[2/n][resources ui] Display nested resources in UI

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Displays nested resources in the Resources UI in a new panel above the config entries.

Screen Shot 2023-03-13 at 10 14 05 AM

## Test Plan

Tested locally.

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #12728** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12728" target="_blank">Graphite</a> 
    * **PR #12729** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12729" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12729?utm_source=stack-comment).

File: js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx [hunk 1]
```diff
@@ -106,65 +128,117 @@ export const ResourceRoot: React.FC<Props> = (props) => {
...
-                    </Table>
+                          </thead>
+                          <tbody>
+                            {Object.keys(nestedResources).map((key) => {
+                              return (
+                                <tr key={key}>
+                                  <td>
+                                    <Box
+                                      flex={{direction: 'column', gap: 4, alignItems: 'flex-start'}}
+                                    >
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx:162]: Box wrapper necessary?

File: js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx [hunk 2]
```diff
@@ -106,65 +128,117 @@ export const ResourceRoot: React.FC<Props> = (props) => {
...
+                    <Box>
+                      <SectionHeader>
+                        <Subheading>Configuration</Subheading>
+                      </SectionHeader>
+                      <Table>
+                        <thead>
+                          <tr>
+                            <th style={{width: 120}}>Key</th>
+                            <th style={{width: 90}}>Type</th>
+                            <th style={{width: 90}}>Value</th>
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx:195]: Are these widths necessary? In the screenshot I notice that the table doesn't take up the entire width of its container.

File: js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx [hunk 3]
```diff
@@ -106,65 +128,117 @@ export const ResourceRoot: React.FC<Props> = (props) => {
...
+                          {resourceDetailsOrError.configFields.map((field) => {
+                            const defaultValue = field.defaultValueAsJson;
+                            const type =
+                              field.name in configuredValues
+                                ? configuredValues[field.name].type
+                                : null;
+                            const actualValue =
+                              field.name in configuredValues
+                                ? configuredValues[field.name].value
+                                : defaultValue;
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx:208]: See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn.

File: js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx [hunk 4]
```diff
@@ -106,65 +128,117 @@ export const ResourceRoot: React.FC<Props> = (props) => {
...
+                                  >
+                                    <strong>{field.name}</strong>
+                                    <div style={{fontSize: 12, color: Colors.Gray700}}>
+                                      {field.description}
+                                    </div>
+                                  </Box>
+                                </td>
+                                <td>{remapName(field.configTypeKey)}</td>
+                                <td>
+                                  <Box flex={{direction: 'row', justifyContent: 'space-between'}}>
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx:225]: `space-between` could end up a little strange here if there is a lot of space for the cell. Maybe just a `gap` value?

File: js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx [hunk 5]
```diff
@@ -238,9 +316,33 @@ const RESOURCE_ROOT_QUERY = gql`
...
+const SectionHeader = styled.div`
+  background-color: ${Colors.Gray50};
+  border: 0;
+  box-shadow: inset 0px -1px 0 ${Colors.KeylineGray}, inset 0px 1px 0 ${Colors.KeylineGray};
+  cursor: pointer;
+  display: block;
+  width: 100%;
+  margin: 0;
+  height: ${SECTION_HEADER_HEIGHT}px;
+  line-height: ${SECTION_HEADER_HEIGHT}px;
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx:344]: Are these necessary?

File: js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx [hunk 6]
```diff
@@ -238,9 +316,33 @@ const RESOURCE_ROOT_QUERY = gql`
...
     }
   }
   ${PYTHON_ERROR_FRAGMENT}
 `;
+
+const SECTION_HEADER_HEIGHT = 48;
+const SectionHeader = styled.div`
+  background-color: ${Colors.Gray50};
+  border: 0;
+  box-shadow: inset 0px -1px 0 ${Colors.KeylineGray}, inset 0px 1px 0 ${Colors.KeylineGray};
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx:338]: You could probably just use a `Box` for this.

File: js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx [hunk 7]
```diff
@@ -238,9 +316,33 @@ const RESOURCE_ROOT_QUERY = gql`
...
+  cursor: pointer;
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx:339]: Does the `SectionHeader` have any interactive behavior? If not, it should be fine to leave the cursor as the default.

**Review by hellendag — CHANGES_REQUESTED** [review]
Some JS requests. :)

File: js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx (hunk 2 above)
Comment (benpankow) [reply js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx:195]: Matter of preference more than anything, it means the columns nicely line up on subsequent tables & widths don't change when navigating from one resource to another which I found jarring

Screen Shot 2023-03-13 at 1 33 33 PM
Screen Shot 2023-03-13 at 1 33 46 PM

File: js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx (hunk 5 above)
Comment (benpankow) [reply js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx:344]: Switched to using Box instead

File: js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx (hunk 2 above)
Comment (hellendag) [reply js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx:201]: Ahh, I see. If just the "Resource" (Table A) and "Type" (Table B) columns need to align, you could just put a pixel width on the first column of each table, and let the browser do the rest.

**Review by hellendag — APPROVED** [review]

**Comment by benpankow** [comment]
@benpankow queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12729).

**Comment by benpankow** [comment]
[Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12729) couldn't merge this PR because it had conflicts with the trunk branch.

**Comment by benpankow** [comment]
@benpankow queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12729).

**Pull request merged**